source/includes/fundamentals/code-snippets/Cursor.java

```diff
@@ -0,0 +1,113 @@
+package docs;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.*;
+
+import org.bson.conversions.Bson;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.bson.Document;
+import com.mongodb.client.model.Filters;
+
+public class Cursor {
+    private final MongoClient mongoClient;
+    private final MongoDatabase database;
+    private MongoCollection<Document> collection;
+
+    private Cursor(){
+        final String uri = System.getenv("DRIVER_REF_URI");
+
+        mongoClient = MongoClients.create(uri);
+        database = mongoClient.getDatabase("test");
+        collection = database.getCollection("orders");
+    }
+
+    public static void main(String [] args){
+        Cursor c = new Cursor();
+        c.setupPaintCollection();
+
+        System.out.println("For Each Iteration");
+        c.forEachIteration();
+
+        System.out.println("First Example");
+        c.firstExample();
+
+        System.out.println("Into Example");
+        c.intoExample();
+
+        System.out.println("Manual Iteration");
+        c.manualIteration();
+        
+        c.closeExample();
+    }
+
+    private void forEachIteration(){
+        // begin forEachIteration
+        FindIterable<Document> iterable = collection.find();
+        iterable.forEach(doc -> System.out.println(doc.toJson()));
+        // end forEachIteration
+    }
+
+    private void firstExample(){
+        // begin firstExample
+        FindIterable<Document> iterable = collection.find();
+        System.out.println(iterable.first());
+        // end firstExample
+    }
+
+    private void intoExample(){
+        // begin intoExample
+        List<Document> results = new ArrayList<>();
+        FindIterable<Document> iterable = collection.find();
+        iterable.into(results);
+        System.out.println(results);
+        // end intoExample
+    }
+
+    private void manualIteration(){
```

Comment:
This example should use a finally block in which `close()` is called.  It's never correct to use a cursor as it is in this example, since the call to `close()` will be skipped if there's an exception